src/api/apiService.js

```diff
@@ -0,0 +1,48 @@
+import { API_URL } from '../constants/constants';
+
+const handleErrors = (response) => {
+  return new Promise((resolve, reject) => {
+    if (!response) {
+      reject({ error: 'No response returned from fetch' });
+      return;
+    }
+
+    if (response.ok) {
+      resolve(response);
+      return;
+    }
+
+    response.json()
+      .then((json) => {
+        const error = json || { error: response.statusText };
+        reject(error);
+      }).catch(() => reject({ error: 'Response not JSON' }));
+  });
+};
+
+class Api {
+  static makeRequest(uri, requestData = {}) {
+    const url = API_URL + uri;
```

Comment:
Isn't this throwing a warning https://eslint.org/docs/rules/prefer-template?

Reply:
No, it throws that rule when you add a string, but not only variables (for example: `const url = API_URL + 'hello';`). That's why I didn't use it, are you using template literals for all cases?